site/_posts/2020-11-13-securing-flutter-oauth.md

```diff
@@ -0,0 +1,766 @@
+
+Then navigate to the "OAuth" tab and add in a redirect URI of `com.fusionauth.flutterdemo://login-callback`. We’ll use this redirect URL in our Flutter application later. Make sure the "Require authentication" checkbox is set to off. Since the Flutter application is a native app, it's not considered a "confidential client" and a client secret isn't safe. It could be found by decompiling the application. The library we will use implements another method of securing access: proof of key code exchange, or PKCE (often pronounced PKCE).
+
+Ensure that the authorization code grant is enabled as well. Add the value of `7e3637e8-723a-42d6-9d1d-5cb36128d6f1` into the `Id` field. If you don't provide one, it will be generated for you. The Id of the application is also the "Client Id"; we'll need that later in the configuration process. 
+
+Then, we need to click "Save". When properly configured, the application details screen should look like this:
+
+{% include _image.liquid src="/assets/img/blogs/securing-flutter-app/application-configuration.png" alt="Configuring the FusionAuth application." class="img-fluid" figure=false %}
+
+We need to make sure that we register our users to the new FusionAuth application. Doing so creates a relationship between a user and the newly created application as shown in the screenshot below:
+
+{% include _image.liquid src="/assets/img/blogs/securing-flutter-app/user-registration.png" alt="Registering the user for the new application." class="img-fluid" figure=false %}
+
+If we want, we can add more users to the "Users" and register them to this application. But you can work through this tutorial with just one registered user. 
+
+If you want to test with a real device, you should also set up ngrok to point to the local FusionAuth server, so that devices can access `http://localhost:9011`. While beyond the scope of this tutorial, here's [ngrok documentation](https://ngrok.com/docs#getting-started-expose) on how to do so. 
+
+Next, we move on to setting up the Flutter app project.
+
+## Setting up the Flutter project
+
+Setting up the Flutter project is simple and easy for every OS. We can follow the instructions in the [Flutter documentation](https://flutter.dev/docs/get-started/install). But to make things easier, a quick outline of the setup steps is provided below.
+
+First, we need to clone the Flutter SDK from GitHub by running the following command in our device terminal:
+
+```shell
+git clone https://github.com/flutter/flutter.git
+```
+
+Note that we need to choose an appropriate directory in which to clone the SDK.
+
+Next, we need to add the flutter tool to our path by running the following command in the terminal (and if you are doing this more than once, perhaps to your `.bashrc` or `.zshrc`):
+
+```shell
+export PATH="$PATH:`pwd`/flutter/bin"
```

Comment:
It's somewhat uncommon to use `pwd` in a PATH assignment since login shells can be opened at any location. This should really be something like `${HOME}/dev/flutter/bin` and the Git clone above should use this path.